In [1]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("hugomathien/soccer")

print("Path to dataset files:", path)

Path to dataset files: C:\Users\yorri\.cache\kagglehub\datasets\hugomathien\soccer\versions\10


In [2]:
import sqlite3
import pandas as pd
import os

# Create the full path to the sqlite file
db_path = os.path.join(path, "database.sqlite")

# Connect to the database
conn = sqlite3.connect(db_path)

# Check what tables are inside
tables = pd.read_sql("SELECT name FROM sqlite_master WHERE type='table';", conn)
print(tables)

                name
0    sqlite_sequence
1  Player_Attributes
2             Player
3              Match
4             League
5            Country
6               Team
7    Team_Attributes


In [3]:
print("DataFrames:")
for table_name in tables['name']:
    query = f"SELECT * FROM {table_name}"
    lower_var_name = table_name.lower()
    print(f" - {lower_var_name}")
    # This creates a variable in your environment named after the table
    globals()[lower_var_name] = pd.read_sql(query, conn)

DataFrames:
 - sqlite_sequence
 - player_attributes
 - player
 - match
 - league
 - country
 - team
 - team_attributes


In [4]:
sqlite_sequence
# Er zijn in totaal 7 rijen en 2 kolommen
# Deze tabel registreert de tabel-IDs van de tabellen in de database. 
        # name = naam van de tabel (string)
        # seq = aantal rijen (int)
# Het vertelt niet hoeveel rijen er in de tabel zitten, maar de hoogst uitgegeven ID-waarde in die tabel.

,name,seq
0,Team,103916
1,Country,51958
2,League,51958
3,Match,51958
4,Player,11075
5,Player_Attributes,183978
6,Team_Attributes,1458


In [5]:
pd.set_option('display.max_columns', None)
import pandas as pd

def check_kolom(df, kolom_naam):
    """Functie om informatie te krijgen over een kolom, om hem te kunnen beschrijven"""
    print(kolom_naam)
    print(f"Datatype: {df[kolom_naam].dtype}")
    print(f"Missende waardes: {df[kolom_naam].isnull().sum()}")

    if pd.api.types.is_numeric_dtype(df[kolom_naam]):
        print(f"Maximale waarde: {df[kolom_naam].max()}")
        print(f"Minimale waarde: {df[kolom_naam].min()}")
    else:
        print("Maximale waarde: N.v.t. (tekstkolom)")
        print("Minimale waarde: N.v.t. (tekstkolom)")

    print("\nWaarde tellingen:")
    print(df[kolom_naam].value_counts().head(10))

check_kolom(player_attributes, 'gk_reflexes')

display(player_attributes.head(5))
# display(player_attributes.info())
# Er zijn in totaal 183978 rijen en 42 kolommen, er missen in 30 kolommen 836 waardes. Er zijn ook nog 7 kolommen waarbij 2713 waardes missen. 
# De kolom attacking_work_rate heeft de meeste missende waardes, namenlijk: 3230.
        # id = de ID van de speler in de tabel (int: 1-183978)
        # player_fifa_api_id = de ID van de speler in FIFA (int: 2-234141)
        # player_api_id = de ID van de speler in the Open Player Database (int: 2625-750584)
        # date = de datum waarop de attributen van de speler zijn opgehaald (object: 2007/02/22-2016/07/07)
        # overall_rating = de gemiddelde rating van de speler (float: 33-94)
        # potential = de maximaal haalbare rating van de speler in de toekomst (float: 39-97)
        # preferred_foot = de voorkeur voet van de speler (object: left, right)
        # attacking_work_rate = hoe hard de speler aanvalt (object: low, medium, high, None, norm, y, le, stoc)
        # defensive_work_rate = hoe hard de speler verdedigt (object: low, medium, high, _0, o, 1, ormal, 2, 3, 5, 7, 6, 0, 9, 4, es, ean, tocky, 8)
        # crossing = de score voor de voorzetten van de speler (float: 1-95)
        # finishing = de score voor de nauwkeurigheid van schoten op doel van de speler(float: 1-97)
        # heading_accuracy = de score voor de nauwkeurigheid van het koppen van de speler (float: 1-98)
        # short_passing = de score voor de nauwkeurigheid van de korte passen van de speler (float: 3-97)
        # volleys = de score voor het vermogen van de speler om een bal vanuit de lucht direct te schieten (float: 1-93)
        # dribbling = de score voor de nauwkeurigheid van de dribbling van de speler (float: 1-97)
        # curve = de score voor het vermogen om de bal met een boog/effect te trappen (float: 2-94)
        # free_kick_accuracy = de score voor de nauwkeurigheid van de speler bij het nemen van directe vrije trappen (float: 1-97)
        # long_passing = de score voor de nauwkeurigheid van de lange passen van de speler (float: 3-97)
        # ball_control = de score voor het vermogen van de speler om de bal onder controle te houden (float: 5-97)
        # acceleration = de score voor hoe snel de speler zijn maximale snelheid bereikt (float: 10-97)
        # sprint_speed = de score voor de maximale loopsnelheid van de speler (float: 12-97)
        # agility = de score voor de wendbaarheid van de speler bij het draaien en veranderen van richting (float: 11-96)
        # reactions = de score voor hoe snel de speler reageert op situaties op het veld (float: 17-97)
        # balance = de score voor het vermogen van de speler om zijn evenwicht te bewaren (float: 12-96)
        # shot_power = de score voor de kracht van het schot van de speler (float: 2-97)
        # jumping = de score voor de sprongkracht van de speler (float: 14-96)
        # stamina = de score voor het uithoudingsvermogen van de speler (float: 10-96)
        # strength = de score voor de fysieke kracht van de speler (float: 10-96)
        # long_shots = de score voor de nauwkeurigheid van schoten vanaf buiten het strafschopgebied (float: 10-96)
        # aggression = de score voor de mate van aggresie bij de speler (float: 6-97)
        # interceptions = de score voor het vermogen om passes te lezen en te onderscheppen van de speler (float: 1-96)
        # positioning = de score voor het vermogen van de speler om zich op de juist positie te positioneren (float: 2-96)
        # vision = de score voor het spelinzicht en overzicht om vrije medespelers en ruimtes te zien van de speler (float: 1-97)
        # penalties = de score voor de nauwkeurigheid van het nemen van penalties van de speler (float: 2-96)
        # marking = de score voor het vermogen om een tegenstander af te dekken en te volgen (float: 1-96)
        # standing_tackle = de score voor de nauwkeurigheid van de staande tackles van de speler (float: 1-95)
        # sliding_tackle = de score voor de nauwkeurigheid van de slide tackles van de speler (float: 2-95)
        # gk_diving = de score voor het vermogen van de speler om naar de hoeken te duiken en schoten te stoppen (float: 1-94)
        # gk_handling = de score voor het vermogen van de speler om de bal klemvast te pakken en vast te houden (float: 1-93)
        # gk_kicking = de score voor de afstand en nauwkeurigheid van de uittrap en passes van de de speler (float: 1-97)
        # gk_positioning = de score voor het vermogen van de speler om de juiste positie in het doel te kiezen (float: 1-96)
        # gk_reflexes = de score voor de reactiesnelheid van de speler op schoten op doel (float: 1-96)
# Statistieken van spelers, overgenomen uit de FIFA-gameserie (EA Sports)
# Aangezien er heel veel kolommen dezelfde hoeveelheid rijen met missende waarde hebben, zouden we de rijen met de missende waardes kunnen verwijderen



gk_reflexes
Datatype: float64
Missende waardes: 836
Maximale waarde: 96.0
Minimale waarde: 1.0

Waarde tellingen:
gk_reflexes
8.0     14156
10.0    14026
14.0    13908
7.0     13886
9.0     13721
13.0    13277
12.0    13043
6.0     12706
11.0    12704
15.0    10594
Name: count, dtype: int64


,id,player_fifa_api_id,player_api_id,date,overall_rating,potential,preferred_foot,attacking_work_rate,defensive_work_rate,crossing,finishing,heading_accuracy,short_passing,volleys,dribbling,curve,free_kick_accuracy,long_passing,ball_control,acceleration,sprint_speed,agility,reactions,balance,shot_power,jumping,stamina,strength,long_shots,aggression,interceptions,positioning,vision,penalties,marking,standing_tackle,sliding_tackle,gk_diving,gk_handling,gk_kicking,gk_positioning,gk_reflexes
0,1,218353,505942,2016-02-18 00:00:00,67.0,71.0,right,medium,medium,49.0,44.0,71.0,61.0,44.0,51.0,45.0,39.0,64.0,49.0,60.0,64.0,59.0,47.0,65.0,55.0,58.0,54.0,76.0,35.0,71.0,70.0,45.0,54.0,48.0,65.0,69.0,69.0,6.0,11.0,10.0,8.0,8.0
1,2,218353,505942,2015-11-19 00:00:00,67.0,71.0,right,medium,medium,49.0,44.0,71.0,61.0,44.0,51.0,45.0,39.0,64.0,49.0,60.0,64.0,59.0,47.0,65.0,55.0,58.0,54.0,76.0,35.0,71.0,70.0,45.0,54.0,48.0,65.0,69.0,69.0,6.0,11.0,10.0,8.0,8.0
2,3,218353,505942,2015-09-21 00:00:00,62.0,66.0,right,medium,medium,49.0,44.0,71.0,61.0,44.0,51.0,45.0,39.0,64.0,49.0,60.0,64.0,59.0,47.0,65.0,55.0,58.0,54.0,76.0,35.0,63.0,41.0,45.0,54.0,48.0,65.0,66.0,69.0,6.0,11.0,10.0,8.0,8.0
3,4,218353,505942,2015-03-20 00:00:00,61.0,65.0,right,medium,medium,48.0,43.0,70.0,60.0,43.0,50.0,44.0,38.0,63.0,48.0,60.0,64.0,59.0,46.0,65.0,54.0,58.0,54.0,76.0,34.0,62.0,40.0,44.0,53.0,47.0,62.0,63.0,66.0,5.0,10.0,9.0,7.0,7.0
4,5,218353,505942,2007-02-22 00:00:00,61.0,65.0,right,medium,medium,48.0,43.0,70.0,60.0,43.0,50.0,44.0,38.0,63.0,48.0,60.0,64.0,59.0,46.0,65.0,54.0,58.0,54.0,76.0,34.0,62.0,40.0,44.0,53.0,47.0,62.0,63.0,66.0,5.0,10.0,9.0,7.0,7.0


In [6]:
pd.set_option('display.max_columns', None)
import pandas as pd

def check_kolom(df, kolom_naam):
    """Functie om informatie te krijgen over een kolom, om hem te kunnen beschrijven"""
    print(kolom_naam)
    print(f"Datatype: {df[kolom_naam].dtype}")
    print(f"Missende waardes: {df[kolom_naam].isnull().sum()}")

    if pd.api.types.is_numeric_dtype(df[kolom_naam]):
        print(f"Maximale waarde: {df[kolom_naam].max()}")
        print(f"Minimale waarde: {df[kolom_naam].min()}")
    else:
        print("Maximale waarde: N.v.t. (tekstkolom)")
        print("Minimale waarde: N.v.t. (tekstkolom)")

    print("\nWaarde tellingen:")
    print(df[kolom_naam].value_counts().head(10))

check_kolom(player, 'weight')

display(player.head(5))
player.info()

# Deze tabel bevat de persoonlijke basisgegevens van alle spelers in de database.
        # id = het unieke rij-nummer in deze tabel (int: 1-11075)
        # player_api_id = de ID van de speler in the Open Player Database (int: 2625-750584)
        # player_name = de naam van de speler (object)
        # player_fifa_api_id = de ID van de speler in FIFA (int: 2-234141)
        # birthday = de geboortedatum van de speler (object: 1967/01/23-1999/04/24)
        # height = de lengte van de speler in cm (float: 157.48-208.28)
        # weight = het gewicht van de speler in lbs (int: 117-243)

weight
Datatype: int64
Missende waardes: 0
Maximale waarde: 243
Minimale waarde: 117

Waarde tellingen:
weight
165    811
176    697
154    660
172    656
168    600
159    595
163    577
161    565
170    551
174    488
Name: count, dtype: int64


,id,player_api_id,player_name,player_fifa_api_id,birthday,height,weight
0,1,505942,Aaron Appindangoye,218353,1992-02-29 00:00:00,182.88,187
1,2,155782,Aaron Cresswell,189615,1989-12-15 00:00:00,170.18,146
2,3,162549,Aaron Doran,186170,1991-05-13 00:00:00,170.18,163
3,4,30572,Aaron Galindo,140161,1982-05-08 00:00:00,182.88,198
4,5,23780,Aaron Hughes,17725,1979-11-08 00:00:00,182.88,154


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11060 entries, 0 to 11059
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   id                  11060 non-null  int64  
 1   player_api_id       11060 non-null  int64  
 2   player_name         11060 non-null  object 
 3   player_fifa_api_id  11060 non-null  int64  
 4   birthday            11060 non-null  object 
 5   height              11060 non-null  float64
 6   weight              11060 non-null  int64  
dtypes: float64(1), int64(4), object(2)
memory usage: 605.0+ KB


In [19]:
pd.set_option('display.max_columns', None)
import pandas as pd

def check_kolom(df, kolom_naam):
    """Functie om informatie te krijgen over een kolom, om hem te kunnen beschrijven"""
    print(kolom_naam)
    print(f"Datatype: {df[kolom_naam].dtype}")
    print(f"Missende waardes: {df[kolom_naam].isnull().sum()}")

    if pd.api.types.is_numeric_dtype(df[kolom_naam]):
        print(f"Maximale waarde: {df[kolom_naam].max()}")
        print(f"Minimale waarde: {df[kolom_naam].min()}")
    else:
        print("Maximale waarde: N.v.t. (tekstkolom)")
        print("Minimale waarde: N.v.t. (tekstkolom)")

    print("\nWaarde tellingen:")
    print(df[kolom_naam].value_counts().head(10))

check_kolom(match, 'date')
pd.set_option('display.max_rows', None)

display(match.tail(10))
match.info()

match['date'].max()
# Deze tabel bevat de matchgegevens van alle matches in de database. Denk aan team id's, match id's en datums
        # id = het unieke rij-nummer in deze tabel (int: 1-25979) 
        # country_id = de ID van de land waar de match gespeeld werd (int: 1-24558)
        # league_id = de ID van de competitie waar de match gespeeld werd (int: 1-24558)
        # season = het seizoen waar de match gespeeld werd (object: 2008/2009-2015/2016)
        # stage = de stage waar de match gespeeld werd (int: 1-38)
        # date = de datum waar de match gespeeld werd (object: 2008/07/18-2016/05/25)
        # match_api_id = de API-ID van de match (int: 483129-2216672)
        # home_team_api_id = de API-ID van de thuis team (int: 1601-274581)
        # away_team_api_id = de API-ID van de uit team (int: 1-24558)
        # home_team_goal = het aantal goals van de thuis team (int: 0-10)
        # away_team_goal = het aantal goals van de uit team (int: 0-9)
    # home_player_X(getal)= Deze geven de horizontale positie op het veld aan (de breedte) per thuis speler, elke speler heeft een ander getal achter de X (float:)
    # away_player_X(getal)= Deze geven de horizontale positie op het veld aan (de breedte) per uit speler, elke speler heeft een ander getal achter de X (float:)
    # home_player_Y(getal)= Deze geven de verticale positie op het veld aan (de lengte van het veld) per thuis speler, elke speler heeft een ander getal achter de Y (float:)
    # away_player_Y(getal)= Deze geven de verticale positie op het veld aan (de lengte van het veld) per uit speler, elke speler heeft een ander getal achter de Y (float:)
    # home_player_(getal) = de API-ID van de thuis spelers, elke speler heeft een ander getal(int: 1-11075)
    # away_player_(getal) = de API-ID van de uit spelers, elke speler heeft een ander getal(int: 1-11075)
    # goal, shoton, shotoff, foulcommit, card, cross, corner, possession = Deze kolommen bevatten gedetailleerde logboeken van gebeurtenissen tijdens de wedstrijd. De data is opgeslagen in XML-formaat en bevat extra details zoals de minuut van de actie en de betrokken spelers.
    # B365H, B365D,	B365A, BWH,	BWD, BWA, IWH, IWD, IWA, LBH, LBD, LBA,	PSH, PSD, PSA, WHH,	WHD, WHA, SJH, SJD,	SJA, VCH, VCD, VCA,	GBH, GBD, GBA, BSH, BSD, BSA = de quoteringen van verschillende bookmakers (Bet365, Bwin, Interwetten, etc.). Bevat veel NaN waardes, want niet elke bookmaker biedt odds aan voor elke competitie en sommige bookmarkers zijn inmiddels faill|iet. 
match.isna().sum()

date
Datatype: object
Missende waardes: 0
Maximale waarde: N.v.t. (tekstkolom)
Minimale waarde: N.v.t. (tekstkolom)

Waarde tellingen:
date
2009-04-11 00:00:00    60
2015-04-04 00:00:00    55
2009-05-16 00:00:00    54
2013-03-30 00:00:00    54
2010-04-03 00:00:00    53
2009-09-26 00:00:00    53
2010-09-11 00:00:00    52
2011-10-29 00:00:00    51
2011-09-24 00:00:00    51
2009-10-31 00:00:00    50
Name: count, dtype: int64


,id,country_id,league_id,season,stage,date,match_api_id,home_team_api_id,away_team_api_id,home_team_goal,away_team_goal,home_player_X1,home_player_X2,home_player_X3,home_player_X4,home_player_X5,home_player_X6,home_player_X7,home_player_X8,home_player_X9,home_player_X10,home_player_X11,away_player_X1,away_player_X2,away_player_X3,away_player_X4,away_player_X5,away_player_X6,away_player_X7,away_player_X8,away_player_X9,away_player_X10,away_player_X11,home_player_Y1,home_player_Y2,home_player_Y3,home_player_Y4,home_player_Y5,home_player_Y6,home_player_Y7,home_player_Y8,home_player_Y9,home_player_Y10,home_player_Y11,away_player_Y1,away_player_Y2,away_player_Y3,away_player_Y4,away_player_Y5,away_player_Y6,away_player_Y7,away_player_Y8,away_player_Y9,away_player_Y10,away_player_Y11,home_player_1,home_player_2,home_player_3,home_player_4,home_player_5,home_player_6,home_player_7,home_player_8,home_player_9,home_player_10,home_player_11,away_player_1,away_player_2,away_player_3,away_player_4,away_player_5,away_player_6,away_player_7,away_player_8,away_player_9,away_player_10,away_player_11,goal,shoton,shotoff,foulcommit,card,cross,corner,possession,B365H,B365D,B365A,BWH,BWD,BWA,IWH,IWD,IWA,LBH,LBD,LBA,PSH,PSD,PSA,WHH,WHD,WHA,SJH,SJD,SJA,VCH,VCD,VCA,GBH,GBD,GBA,BSH,BSD,BSA
25969,25970,24558,24558,2015/2016,8,2015-09-12 00:00:00,1992086,9931,10190,2,1,1.0,2.0,4.0,6.0,8.0,4.0,6.0,3.0,5.0,7.0,5.0,1.0,2.0,4.0,6.0,8.0,5.0,2.0,4.0,6.0,8.0,5.0,1.0,3.0,3.0,3.0,3.0,6.0,6.0,8.0,8.0,8.0,11.0,1.0,3.0,3.0,3.0,3.0,6.0,8.0,8.0,8.0,8.0,11.0,156175.0,95216.0,172768.0,22834.0,30835.0,339992.0,114738.0,176206.0,34035.0,40274.0,23148.0,42231.0,678384.0,638592.0,67363.0,413155.0,45780.0,143790.0,66934.0,176298.0,92094.0,72628.0,None,None,None,None,None,None,None,None,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25970,25971,24558,24558,2015/2016,8,2015-09-12 00:00:00,1992087,10192,9824,4,0,1.0,2.0,4.0,6.0,8.0,2.0,4.0,6.0,8.0,4.0,6.0,1.0,2.0,4.0,6.0,8.0,3.0,5.0,7.0,3.0,5.0,7.0,1.0,3.0,3.0,3.0,3.0,7.0,7.0,7.0,7.0,10.0,10.0,1.0,3.0,3.0,3.0,3.0,7.0,7.0,7.0,10.0,10.0,10.0,274787.0,492132.0,108451.0,25815.0,94553.0,384376.0,598355.0,320184.0,45174.0,71764.0,302079.0,33272.0,114735.0,25813.0,41621.0,358156.0,114011.0,42237.0,56868.0,209371.0,531309.0,113227.0,None,None,None,None,None,None,None,None,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25971,25972,24558,24558,2015/2016,8,2015-09-13 00:00:00,1992088,10199,9956,3,3,1.0,2.0,4.0,6.0,8.0,5.0,4.0,6.0,5.0,4.0,6.0,1.0,2.0,4.0,6.0,8.0,4.0,6.0,3.0,5.0,7.0,5.0,1.0,3.0,3.0,3.0,3.0,6.0,7.0,7.0,9.0,11.0,11.0,1.0,3.0,3.0,3.0,3.0,6.0,6.0,8.0,8.0,8.0,11.0,42276.0,27232.0,178119.0,150007.0,42294.0,73977.0,114792.0,201704.0,36382.0,104646.0,34082.0,132814.0,274779.0,155075.0,294256.0,42258.0,173936.0,39979.0,147959.0,451983.0,80564.0,289472.0,None,None,None,None,None,None,None,None,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25972,25973,24558,24558,2015/2016,8,2015-09-13 00:00:00,1992089,10243,10191,3,3,1.0,2.0,4.0,6.0,8.0,2.0,4.0,6.0,8.0,4.0,6.0,1.0,2.0,4.0,6.0,8.0,4.0,6.0,3.0,5.0,7.0,5.0,1.0,3.0,3.0,3.0,3.0,7.0,7.0,7.0,7.0,10.0,10.0,1.0,3.0,3.0,3.0,3.0,6.0,6.0,8.0,8.0,8.0,11.0,274776.0,451982.0,34268.0,197757.0,121080.0,462608.0,67008.0,632356.0,198082.0,194532.0,42262.0,67311.0,8800.0,320183.0,67304.0,67389.0,133126.0,186524.0,93223.0,121115.0,25809.0,289732.0,None,None,None,None,None,None,None,None,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25973,25974,24558,24558,2015/2016,8,2015-09-13 00:00:00,1992090,10179,7896,3,0,1.0,2.0,4.0,6.0,8.0,4.0,6.0,3.0,5.0,7.0,5.0,1.0,2.0,4.0,6.0,8.0,5.0,4.0,6.0,3.0,5.0,7.0,1.0,3.0,3.0,3.0,3.0,6.0,6.0,8.0,8.0,8.0,11.0,1.0,3.0,3.0,3.0,3.0,6.0,8.0,8.0,10.0,10.0,10.0,10637.0,67349.0,202663.0,3259

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25979 entries, 0 to 25978
Columns: 115 entries, id to BSA
dtypes: float64(96), int64(9), object(10)
memory usage: 22.8+ MB


id                      0
country_id              0
league_id               0
season                  0
stage                   0
date                    0
match_api_id            0
home_team_api_id        0
away_team_api_id        0
home_team_goal          0
away_team_goal          0
home_player_X1       1821
home_player_X2       1821
home_player_X3       1832
home_player_X4       1832
home_player_X5       1832
home_player_X6       1832
home_player_X7       1832
home_player_X8       1832
home_player_X9       1832
home_player_X10      1832
home_player_X11      1832
away_player_X1       1832
away_player_X2       1832
away_player_X3       1832
away_player_X4       1832
away_player_X5       1832
away_player_X6       1832
away_player_X7       1832
away_player_X8       1832
away_player_X9       1833
away_player_X10      1833
away_player_X11      1839
home_player_Y1       1821
home_player_Y2       1821
home_player_Y3       1832
home_player_Y4       1832
home_player_Y5       1832
home_player_

In [8]:
print("Row count for each DataFrame:")
for name in tables['name']:
    name = name.lower()
    df = globals()[name]
    print(name, ":", len(df))

Row count for each DataFrame:
sqlite_sequence : 7
player_attributes : 183978
player : 11060
match : 25979
league : 11
country : 11
team : 299
team_attributes : 1458


In [18]:
for table in tables['name']:
    table = table.lower()
    print(40 * "-")
    print("↓ Info for ", table, "↓")
    print(40 * "-")
    print(globals()[table].describe())

----------------------------------------
↓ Info for  sqlite_sequence ↓
----------------------------------------
                 seq
count       7.000000
mean    65185.857143
std     62082.942398
min      1458.000000
25%     31516.500000
50%     51958.000000
75%     77937.000000
max    183978.000000
----------------------------------------
↓ Info for  player_attributes ↓
----------------------------------------
                 id  player_fifa_api_id  player_api_id  overall_rating  \
count  183978.00000       183978.000000  183978.000000   183142.000000   
mean    91989.50000       165671.524291  135900.617324       68.600015   
std     53110.01825        53851.094769  136927.840510        7.041139   
min         1.00000            2.000000    2625.000000       33.000000   
25%     45995.25000       155798.000000   34763.000000       64.000000   
50%     91989.50000       183488.000000   77741.000000       69.000000   
75%    137983.75000       199848.000000  191080.000000       73.000

In [10]:
# Zoek FC Barcelona in de team tabel
barca = team[team['team_long_name'] == 'FC Barcelona']
display(barca[['team_api_id', 
             'team_fifa_api_id', 
             'team_long_name', 
             'team_short_name']])

# Zoek de league_id voor Spanje
spain_league = league[league['name'].str.contains("Spain")]
spain_id = spain_league['id'].values[0]

print(f"De ID voor de Spaanse La Liga is: {spain_id}")

,team_api_id,team_fifa_api_id,team_long_name,team_short_name
258,8634,241.0,FC Barcelona,BAR


De ID voor de Spaanse La Liga is: 21518


In [11]:
import pandas as pd

class LeagueTable:
    def __init__(self, matches_df, teams_df):
        """Initializes the calculator with match and team data."""
        self.matches = matches_df
        self.teams = teams_df

    def _determine_points(self, row):
        """Internal logic to calculate points for a single match."""
        if row['home_team_goal'] > row['away_team_goal']:
            return 3, 0  # Home winstkans
        elif row['home_team_goal'] < row['away_team_goal']:
            return 0, 3  # Away win
        else:
            return 1, 1  # Draw

    def get_standings(self, league_id, season):
        """Processes the data and returns a sorted standings DataFrame."""
        # 1. Filter data
        season_matches = self.matches[
            (self.matches['league_id'] == league_id) & 
            (self.matches['season'] == season)
        ].copy()

        # 2. Assign points
        point_results = season_matches.apply(
            lambda x: pd.Series(self._determine_points(x)), axis=1
        )
        season_matches[['home_points', 'away_points']] = point_results

        # 3. Aggregate points
        home_results = season_matches.groupby('home_team_api_id')['home_points'].sum()
        away_results = season_matches.groupby('away_team_api_id')['away_points'].sum()
        total_points = (home_results + away_results).reset_index()
        total_points.columns = ['team_api_id', 'points']

        # 4. Merge team names and sort
        standings = total_points.merge(
            self.teams[['team_api_id', 'team_long_name']], 
            on='team_api_id'
        )
        
        return standings.sort_values(by='points', ascending=False).reset_index(drop=True)

# --- Usage ---
calculator = LeagueTable(match, team)
la_liga_2016 = calculator.get_standings(spain_id, '2015/2016')
display(la_liga_2016[['team_long_name', 'points']])

,team_long_name,points
0,FC Barcelona,91
1,Real Madrid CF,90
2,Atlético Madrid,88
3,Villarreal CF,64
4,Athletic Club de Bilbao,62
5,RC Celta de Vigo,60
6,Sevilla FC,52
7,Real Sociedad,48
8,Málaga CF,48
9,Real Betis Balompié,45


# Data analyse

## **sqlite_sequence**

Er zijn in totaal **7 rijen** aan systeemgegevens in deze tabel te vinden. De dataset is volledig compleet, er zijn **geen missende waardes** aanwezig.

### Algemene informatie

De tabel `sqlite_sequence` is een automatisch gegenereerde beheer tabel binnen een SQLite-database die een specifieke technische rol heeft. Het zorgen voor unieke identificatienummers.

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Identificatie** | `name` | `text` | De specifieke naam van de tabel in de database. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Teller** | `seq` | `int64` | De hoogst uitgegeven ID-waarde (sequentie) in die specifieke tabel. | 0 | 1.458 | 183.978 | 65.185,9 | 51.958,0 |

---

### Relaties

**sqlite_sequence**

De tabel sqlite_sequence heeft geen connectie met de andere tabellen. Dit komt omdat deze geen data bevat die een relatie heeft met voetbal, maar een interne systeemtabel van de SQLite database zelf is. Deze tabel wordt op de achtergrond gebruikt om ID-nummers bij te houden en is daarom niet gekoppeld aan de andere tabellen.


## **Player Attributes**

Er zijn in totaal **183.978 rijen** aan spelerstatistieken. De dataset is zeer uitgebreid, maar er zijn: 30 kolommen missen 836 waardes, en bij 7 kolommen missen er zelfs 2.713. De kolom `attacking_work_rate` heeft met 3.230 de meeste missende waardes.

### Algemene informatie

De tabel `player attributes` is de belangrijkste plek voor scores van de voetballers. Dankzij 3 verschillende Id nummers kan je dit tabel makkelijk koppelen aan de gespeelde wedstrijden van een speler en persoonlijke gegevens. 

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Systeem** | `id` | `int64` | Interne ID van de rij. | 0 | 1 | 183.978 | 91.989,5 | 91.989,5 |
|  | `player_fifa_api_id` | `int64` | Officiële FIFA-referentie ID. | 0 | 2 | 234.141 | 165.671,5 | 183.488,0 |
|  | `player_api_id` | `int64` | ID in de Open Player Database. | 0 | 2.625 | 750.584 | 135.900,6 | 77.741,0 |
|  | `date` | `object` | Datum van de scouting/update. | 0 | 2007-02-22 | 2016-07-07 | Nvt | Nvt |
| **Algemeen** | `overall_rating` | `float64` | De huidige algemene rating. | 836 | 33,0 | 94,0 | 68,60 | 69,0 |
|  | `potential` | `float64` | Het plafond van de speler voor de toekomst. | 836 | 39,0 | 97,0 | 73,46 | 74,0 |
|  | `preferred_foot` | `object` | Voorkeursvoet (Left / Right). | 836 | Nvt | Nvt | Nvt | Nvt |
|  | `attacking_work_rate` | `object` | Inzet bij aanval (Low, Medium, High, etc.). | 3.230 | Nvt | Nvt | Nvt | Nvt |
|  | `defensive_work_rate` | `object` | Inzet bij verdediging (Low, Medium, High, etc.). | 836 | Nvt | Nvt | Nvt | Nvt |
| **Aanvallend** | `crossing` | `float64` | Kwaliteit van de voorzetten. | 836 | 1,0 | 95,0 | 55,09 | 59,0 |
|  | `finishing` | `float64` | Score voor schotprecisie op doel. | 836 | 1,0 | 97,0 | 49,92 | 53,0 |
|  | `heading_accuracy` | `float64` | Score voor nauwkeurigheid bij koppen. | 836 | 1,0 | 98,0 | 57,27 | 60,0 |
|  | `short_passing` | `float64` | Precisie van korte passes. | 836 | 3,0 | 97,0 | 62,43 | 65,0 |
|  | `volleys` | `float64` | Techniek bij schoten direct uit de lucht. | 2.713 | 1,0 | 93,0 | 49,47 | 52,0 |
| **Techniek** | `dribbling` | `float64` | Controle tijdens het rennen met de bal. | 836 | 1,0 | 97,0 | 59,18 | 64,0 |
|  | `curve` | `float64` | Effect/boog meegeven aan de bal. | 2.713 | 2,0 | 94,0 | 52,97 | 56,0 |
|  | `free_kick_accuracy` | `float64` | Precisie bij directe vrije trappen. | 836 | 1,0 | 97,0 | 49,38 | 50,0 |
|  | `long_passing` | `float64` | Precisie van lange passes. | 836 | 3,0 | 97,0 | 57,07 | 59,0 |
|  | `ball_control` | `float64` | De eerste aanname en controle. | 836 | 5,0 | 97,0 | 63,39 | 67,0 |
| **Fysiek** | `acceleration` | `float64` | Hoe snel een speler op topsnelheid is. | 836 | 10,0 | 97,0 | 67,66 | 69,0 |
|  | `sprint_speed` | `float64` | De maximale loopsnelheid. | 836 | 12,0 | 97,0 | 68,05 | 69,0 |
|  | `agility` | `float64` | Wendbaarheid en draaisnelheid. | 2.713 | 11,0 | 96,0 | 65,97 | 68,0 |
|  | `reactions` | `float64` | Alertheid op situaties.  | 836 | 17,0 | 96,0 | 66,10 | 67,0 |
|  | `balance` | `float64` | Evenwicht tijdens acties. | 2.713 | 12,0 | 96,0 | 65,19 | 67,0 |
| **Kracht** | `shot_power` | `float64` | De hardheid van het schot. | 836 | 2,0 | 97,0 | 61,81 | 65,0 |
|  | `jumping` | `float64` | Sprongkracht. | 2.713 | 14,0 | 96,0 | 66,97 | 68,0 |
|  | `stamina` | `float64` | Uithoudingsvermogen. | 836 | 10,0 | 96,0 | 67,04 | 69,0 |
|  | `strength` | `float64` | Fysieke kracht in duels. | 836 | 10,0 | 96,0 | 67,42 | 69,0 |
| **Mentale** | `long_shots` | `float64` | Precisie van afstandsschoten. | 836 | 1,0 | 96,0 | 53,34 | 58,0 |
|  | `aggression` | `float64` | Mate van agressie en vechtlust. | 836 | 6,0 | 97,0 | 60,95 | 64,0 |
|  | `interceptions` | `float64` | Vermogen om passes te onderscheppen. | 836 | 1,0 | 96,0 | 52,01 | 57,0 |
|  | `positioning` | `float64` | Kiezen van de juiste positie in de aanval. | 836 | 2,0 | 96,0 | 55,79 | 60,0 |
|  | `vision` | `float64` | Overzicht en spelinzicht. | 2.713 | 1,0 | 97,0 | 57,87 | 60,0 |
| **Speciale** | `penalties` | `float64` | Precisie vanaf de penaltystip. | 836 | 2,0 | 96,0 | 55,00 | 57,0 |
|  | `marking` | `float64` | Dekken van de tegenstander. | 836 | 1,0 | 96,0 | 46,77 | 50,0 |
|  | `standing_tackle` | `float64` | Staande de bal afpakken. | 836 | 1,0 | 95,0 | 50,35 | 56,0 |
|  | `sliding_tackle` | `float64` | Glijdende tackle precisie. | 2.713 | 2,0 | 95,0 | 48,00 | 53,0 |
| **Keepers** | `gk_diving` | `float64` | Duiken naar de hoeken. | 836 | 1,0 | 94,0 | 14,70 | 10,0 |
|  | `gk_handling` | `float64` | Klemvast houden van de bal. | 836 | 1,0 | 93,0 | 16,06 | 11,0 |
|  | `gk_kicking` | `float64` | Uittrappen en verre passes. | 836 | 1,0 | 97,0 | 21,00 | 12,0 |
|  | `gk_positioning` | `float64` | Positiekiezen in het doel. | 836 | 1,0 | 96,0 | 16,13 | 11,0 |
|  | `gk_reflexes` | `float64` | Reactiesnelheid op de lijn. | 836 | 1,0 | 96,0 | 16,44 | 11,0 |

---

### Relaties

**Player_Attributes <-> Player**

De tabellen Player en Player_Attributes hebben een connectie, omdat de kwaliteiten en FIFA-statistieken van een speler door de tijd heen veranderen en worden bijgehouden. Deze 2 tabellen zijn gekoppeld, doordat de kolom player_api_id in de Player_Attributes tabel verwijst naar de kolom player_api_id in de Player tabel.

## **Player**

Er zijn in totaal **11.075 unieke spelers** in deze tabel te vinden. De dataset is volledig compleet, er zijn **geen missende waardes** aanwezig in de kolommen.

### Algemene informatie

De tabel `player` bevat de persoonlijke gegevens van alle voetballers. Deze tabel werkt als een centraal punt om namen te koppelen aan api's die je ook bij andere tabellen terugvindt. 

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Relatie** | `id` | `int64` | De primaire sleutel van de rij. | 0 | 1 | 11.075 | 5.537,5 | 5.536,5 |
| **Relatie** | `player_api_id` | `int64` | De unieke identifier voor de Open Player Database. | 0 | 2.625 | 750.584 | 156.582,4 | 96.619,5 |
| **Referentie** | `player_fifa_api_id` | `int64` | De unieke identifier die overeenkomt met de FIFA-games. | 0 | 2 | 234.141 | 165.664,9 | 184.671,0 |
| **Naamgeving** | `player_name` | `object` | De volledige naam van de speler. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Persoonlijk** | `birthday` | `object` | Geboortedatum. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Fysiek** | `height` | `float64` | De lengte van de speler in centimeters. | 0 | 157,48 | 208,28 | 181,87 | 182,88 |
| **Fysiek** | `weight` | `int64` | Het gewicht van de speler in lbs. | 0 | 117 | 243 | 168,38 | 168,0 |

---

### Relaties

**Player <-> Match**

De tabellen Match en Player hebben een connectie, omdat de opstellingen de 11 basisspelers van beide teams (thuis- en uitteam) bij elke wedstrijd geregistreerd staan. Deze 2 tabellen zijn gekoppeld, doordat de speler kolommen zoals: home_player_1-11 en away_player_1-11 in de Match tabel verwijzen naar de kolom player_api_id in de Player tabel.

## **Match**

Er zijn in totaal **25.979 rijen** aan wedstrijdgegevens in deze tabel te vinden. Er zijn verschillende kolommen met missende waardes, zoals de home/away players die rond de ~1.800 zitten. Waar de meeste missende waardes zijn zijn de bookmakers die oplopen tot ~11.000 missende waardes.

### Algemene informatie

De tabel matches is het hart van de database, bijna alle tabellen komen hier wel samen. Dit is de belangrijkste plek kwa identifcatie, wie tegen wie speelde, het land en meer. 

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend (Min - Max) | Min (Min - Max) | Max (Min - Max) | Gem. (Min - Max) | Mediaan (Min - Max) |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Relatie** | `id` | `int64` | Uniek rij-nummer in de tabel. | 0 | 1 | 25.979 | 12.990,0 | 12.990,0 |
| **Relatie** | `country_id` | `int64` | Verwijzing naar het land waar de match gespeeld werd. | 0 | 1 | 24.558 | 11.738,6 | 10.257,0 |
| **Relatie** | `league_id` | `int64` | Verwijzing naar de betreffende competitie. | 0 | 1 | 24.558 | 11.738,6 | 10.257,0 |
| **Context** | `season` | `object` | Het voetbalseizoen. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Context** | `stage` | `int64` | De speelronde van de competitie. | 0 | 1 | 38 | 18,24 | 18,0 |
| **Context** | `date` | `object` | De datum van de match. | 0 | 2008-08-16  | 2016-05-25  | Nvt | Nvt |
| **Relatie** | `match_api_id` | `int64` | De unieke API-ID van de wedstrijd zelf. | 0 | 483.129 | 2.216.672 | 1.195.429 | 1.147.511 |
| **Relatie** | `home_team_api_id` | `int64` | De API-ID van het spelende thuisteam. | 0 | 1.601 | 274.581 | 9.984,4 | 8.697,0 |
| **Relatie** | `away_team_api_id` | `int64` | De API-ID van het spelende uitteam. | 0 | 1.601 | 274.581 | 9.984,5 | 8.697,0 |
| **Uitslag** | `home_team_goal` | `int64` | Aantal doelpunten gescoord door het thuisteam. | 0 | 0 | 10 | 1,54 | 1,0 |
| **Uitslag** | `away_team_goal` | `int64` | Aantal doelpunten gescoord door het uitteam. | 0 | 0 | 9 | 1,16 | 1,0 |
| **Positie X** | `home_player_X1-11` | `float64` | Horizontale positie (breedte) van de 11 thuisspelers. | 1.821 - 1.832 | 0,0 - 2,0 | 2,0 - 9,0 | 1,00 - 7,54 | 1,0 - 8,0 |
| **Positie X** | `away_player_X1-11` | `float64` | Horizontale positie (breedte) van de 11 uitspelers. | 1.832 - 1.839 | 1,0 - 2,0 | 6,0 - 9,0 | 1,00 - 7,53 | 1,0 - 6,0 |
| **Positie Y** | `home_player_Y1-11` | `float64` | Verticale positie (lengte) van de 11 thuisspelers. | 1.821 - 1.832 | 0,0 - 3,0 | 3,0 - 11,00 | 1,00 - 10,44 | 1,0 - 10,0 |
| **Positie Y** | `away_player_Y1-11` | `float64` | Verticale positie (lengte) van de 11 uitspelers. | 1.832 - 1.839 | 1,0 - 7,0 | 3,0 - 11,0 | 1,00 - 10,46 | 1,0 - 10,0 |
| **Spelers** | `home_player_1-11` | `int64` | De API-ID's van de 11 basisspelers van de thuisploeg. | 1.224 - 1.555 | 2.625 - 2.984 | 692.984 - 748.432 | 76.638,4 - 111.131,5 | 38.230,0 - 45.996,0 |
| **Spelers** | `away_player_1-11` | `int64` | De API-ID's van de 11 basisspelers van de uitploeg. |  1.234 - 1.554 | 2.625 - 2.796 | 698.273 - 748.432 | 76.628,2 - 111.087,0 | 38.289,0 - 46.212,0 |
| **Events** | `goal`, `shoton`, `shotoff` | `XML` | Log van goals, schoten op doel en schoten naast. | 11.762 | Nvt | Nvt | Nvt | Nvt |
| **Events** | `foulcommit`, `card` | `XML` | Log van begane overtredingen en gegeven kaarten. | 11.762 | Nvt | Nvt | Nvt | Nvt |
| **Events** | `cross`, `corner` | `XML` | Log van voorzetten en hoekschoppen. | 11.762 | Nvt | Nvt | Nvt | Nvt |
| **Events** | `possession` | `XML` | Gedetailleerde balbezit-percentages per tijdsblok. | 11.762 | Nvt | Nvt | Nvt | Nvt |
| **Betting** | `B365H, B365D, B365A` | `float64` | Odds van **Bet365** (Thuis, Gelijk, Uit). | 3.387 | 1,04 - 1,40 | 17,0 - 51,0 | 2,63 - 4,66 | 2,10 - 3,50 |
| **Betting** | `BWH, BWD, BWA` | `float64` | Odds van **Betway** (Thuis, Gelijk, Uit). | 3.404 | 1,03 - 1,65 | 19,5 - 51,0 | 2,56 - 4,40 | 2,10 - 3,40 |
| **Betting** | `IWH, IWD, IWA` | `float64` | Odds van **Interwetten** (Thuis, Gelijk, Uit). | 3.459 | 1,03 - 1,50 | 11,0 - 25,0 | 2,47 - 4,15 | 2,10 - 3,30 |
| **Betting** | `LBH, LBD, LBA` | `float64` | Odds van **Ladbrokes** (Thuis, Gelijk, Uit). | 3.423 | 1,04 - 1,40 | 19,0 - 51,0 | 2,54 - 4,39 | 2,10 - 3,40 |
| **Betting** | `PSH, PSD, PSA` | `float64` | Odds van **Pinnacle** (Thuis, Gelijk, Uit). | 14.811 | 1,04 - 2,20 | 29,0 - 47,5 | 2,82 - 4,97 | 2,20 - 3,61 |
| **Betting** | `WHH, WHD, WHA` | `float64` | Odds van **William Hill** (Thuis, Gelijk, Uit). | 3.408 | 1,02 - 1,08 | 17,0 - 51,0 | 2,58 - 4,48 | 2,15 - 3,40 |
| **Betting** | `SJH, SJD, SJA` | `float64` | Odds van **Stan James** (Thuis, Gelijk, Uit). | 8.882 | 1,04 - 1,40 | 15,0 - 41,0 | 2,57 - 4,62 | 2,10 - 3,50 |
| **Betting** | `VCH, VCD, VCA` | `float64` | Odds van **VC Bet** (Thuis, Gelijk, Uit). | 3.411 | 1,03 - 1,62 | 26,0 - 67,0 | 2,67 - 4,84 | 2,15 - 3,50 |
| **Betting** | `GBH, GBD, GBA` | `float64` | Odds van **Gamebookers** (Thuis, Gelijk, Uit). | 11.817 | 1,05 - 1,45 | 11,0 - 34,0 | 2,50 - 4,35 | 2,10 - 3,40 |
| **Betting** | `BSH, BSD, BSA` | `float64` | Odds van **Blue Square** (Thuis, Gelijk, Uit). | 11.818 | 1,04 - 1,33 | 13,0 - 34,0 | 2,50 - 4,41 | 2.10 - 3,40 |

---

### Relaties

**Match <-> League**

De tabellen Match en League hebben een connectie, omdat elke wedstrijd onderdeel is van een specifieke competitie. Deze 2 tabellen zijn gekoppeld, doordat de kolom league_id in de Match tabel verwijst naar de kolom id in de League tabel.

**Match <-> Country**

De tabellen Match en Country hebben een connectie, omdat elke wedstrijd in een bepaald land wordt gespeeld. Deze 2 tabellen zijn gekoppeld, doordat de kolom country_id in de Match tabel verwijst naar de kolom id in de Country tabel.

## **League**

Er zijn in totaal **11 rijen** aan competities in deze tabel te vinden. De dataset is volledig compleet, er zijn **geen missende waardes** aanwezig.

### Algemene informatie

De tabel `league` zorgt voor de koppeling tussen de competities en landen. Deze dataset is heel overzichtelijk dat elk land een eigen competitie heeft.

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Relatie** | `id` | `int64` | De primaire sleutel van de rij in deze specifieke tabel. | 0 | 1 | 24.558 | 12.452,1 | 13.274,0 |
| **Relatie** | `country_id` | `int64` | De vreemde sleutel die verwijst naar de `Country` tabel. | 0 | 1 | 24.558 | 12.452,1 | 13.274,0 |
| **Naamgeving** | `name` | `object` | De volledige naam van de competitie. | 0 | Nvt | Nvt | Nvt | Nvt |

---

### Relaties

## **Country**

Er zijn in totaal **11 unieke landen** in deze tabel te vinden. De dataset is volledig compleet, er zijn **geen missende waardes** aanwezig.

### Algemene informatie

De tabel `country` bevat de gegevens van alle landen in de database. De tabel is vooral bedoeld voor identificatie om de landen te koppelen aan id's.

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Relatie** | `id` | `int64` | De primaire sleutel van de rij in deze specifieke tabel. | 0 | 1 | 24.558 | 12.452,1 | 13.274,0 |
| **Naamgeving** | `name` | `object` | De volledige naam van het land. | 0 | Nvt | Nvt | Nvt | Nvt |

---

### Relaties

**Country <-> League**

De tabellen Country en League hebben een connectie met elkaar, omdat elk land een competitie heeft. Deze 2 tabellen zijn gekoppeld, doordat de kolom country_id in de League tabel verwijst naar de kolom id in de Country tabel.

## **Team**

Er zijn in totaal **299 unieke teams** in deze tabel te vinden. De dataset is grotendeels compleet, met uitzondering van  `team_fifa_api_id`. Deze kolom bevatnamelijk 11 missende waardes (zo een 3,7%).

### Algemene informatie

De tabel `team` bevat de gegevens van alle voetbalclubs in de database. Ook deze tabel is bedoeld om de namen van de clubs te koppelen aan id's.

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Relatie** | `id` | `int64` | De primaire sleutel van de rij in deze specifieke tabel. | 0 | 1 | 51.606 | 23.735,3 | 22.805,0 |
| **Relatie** | `team_api_id` | `int64` | De unieke identifier die wordt gebruikt om teams te linken aan wedstrijddata. | 0 | 1.601 | 274.581 | 12.340,5 | 8.655,0 |
| **Referentie** | `team_fifa_api_id` | `float64` | De officiële ID die door FIFA/EA Sports wordt gebruikt. | 11 | 1,0 | 112.513 | 21.534,3 | 673,5 |
| **Naamgeving** | `team_long_name` | `object` | De volledige officiële naam van de voetbalclub. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Naamgeving** | `team_short_name` | `object` | De afkorting van de clubnaam. | 0 | Nvt | Nvt | Nvt | Nvt |

---

### Relaties

**Team <-> Match**

De tabellen Match en Team hebben een connectie met elkaar, omdat aan elke wedstrijd twee teams meedoen: een thuisploeg en een uitploeg. Deze 2 tabellen zijn gekoppeld, doordat de kolommen home_team_api_id en away_team_api_id in de Match tabel verwijzen naar de primaire sleutel team_api_id in de Team tabel.

## **Team Attributes**

Er zijn in totaal **1458 rijen** aan team attributes. De dataset is grotendeels compleet, met uitzondering van `buildUpPlayDribbling`, deze kolom bevat namelijk missende waardes (zo een  66%).

### Algeme informatie

De tabel `team attributes` zorgt voor de statistieken van de clubs en hoe ze spelen.

---

### Kenmerken & Definities

| Categorie | Attribuut | Type | Beschrijving | Missend | Min | Max | Gem. | Mediaan |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| **Systeem** | `id` | `int64` | De primaire sleutel van de rij in deze specifieke tabel. | 0 | 1 | 1.458 | 729,5 | 729,5 |
| | `team_fifa_api_id` | `int64` | De officiële ID die door FIFA/EA Sports wordt gebruikt. | 0 | 1 | 112.513 | 17.707,0 | 485,0 |
| | `team_api_id` | `int64` | De unieke identifier die wordt gebruikt om teams te linken aan wedstrijddata. | 0 | 1.601 | 274.581 | 9.995,7 | 8.674,0 |
| **Opbouw** | `buildUpPlaySpeed` | `int64` | Snelheid van de spelopbouw (0 = Traag, 100 = Snel). | 0 | 20,0 | 80,0 | 52,46 | 52,0 |
|  | `buildUpPlayDribbling` | `float64` | Mate van dribbelen (0 = Weinig, 100 = Veel). | 969 | 24,0 | 77,0 | 48,61 | 49,0 |
|  | `buildUpPlayPassing` | `int64` | Afstand van passing (0 = Kort/Dichtbij, 100 = Lang/Ver). | 0 | 20,0 | 80,0 | 48,49 | 50,0 |
|  | `buildUpPlayPositioning` | `object` | Structuur: **Organised** (vaste posities) of **Free Form**. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Kansen** | `chanceCreationPassing` | `int64` | Kansen met passes (0 = Veilig, 100 = Risicovol/Risky). | 0 | 21,0 | 80,0 | 52,17 | 52,0 
|  | `chanceCreationCrossing` | `int64` | Frequentie van voorzetten (0 = Zelden, 100 = Veel). | 0 | 20,0 | 80,0 | 53,73 | 53,0 |
|  | `chanceCreationShooting` | `int64` | Kansen om op doel te schieten (0 = Weinig, 100 = Veel). | 0 | 22,0 | 80,0 | 53,97 | 53,0 |
|  | `chanceCreationPosition` | `object` | Structuur voorin: **Organised** of **Free Form**. | 0 | Nvt | Nvt | Nvt | Nvt |
| **Defensie** | `defencePressure` | `int64` | Hoogte van druk zetten op verdediging (0 = Deep, 100 = High/Hoog). | 0 | 23,0 | 72,0 | 46,02 | 45,0 |
|  | `defenceAggression` | `int64` | Stijl van verdedigen (0 = Contain/Afwachten, 100 = Double/Agressief). | 0 | 24,0 | 72,0 | 49,25 | 48,0 |
|  | `defenceTeamWidth` | `int64` | Breedte van het verdedigend team (0 = Narrow/Smal, 100 = Wide/Breed). | 0 | 29,0 | 73,0 | 52,19 | 52,0 |
|  | `defenceDefenderLine` | `object` | Strategie: **Cover** (rugdekking) of **Offside Trap** (buitenspel). | 0 | Nvt | Nvt | Nvt | Nvt |

---

### Relaties

**Team_Attributes <->  Team**

De tabellen Team en Team_Attributes hebben een connectie, 
omdat de speelstijl en statistieken van een club per seizoen worden vastgelegd. Deze 2 tabellen zijn gekoppeld, doordat de kolom team_api_id in de Team_Attributes tabel verwijst naar de kolom team_api_id in de Team tabel.